### PR TITLE
ci(dependencies): fix ci by pinning python version

### LINKED ISSUE #1: CI: Fix CI
CI is failing since 3 weeks ago, we need to fix that

### PR DESCRIPTION
**Pull request by tintayadev**
Resolves #1  
## Pull Request description

We have investigated the issue with the ci, and have found that the dependencies were failing to install because the conda environment file (`conda/dev.yaml`) pinned just the minimum python version (>=3.8.1) which caused conda to install python 3.12 causing dependencies issues (mainly numpy 1.25.2 which only supports python 3.9 to 3.11).

## How to test these changes

* ```...```

## Pull Request checklists

This PR is a:
- [ ] bug-fix
- [ ] new feature
- [ ] maintenance

About this PR:
- [ ] it includes tests.
- [ ] the tests are executed on CI.
- [ ] the tests generate log file(s) (path).
- [x] pre-commit hooks were executed locally.
- [ ] this PR requires a project documentation update.

Author's checklist:
- [x] I have reviewed the changes and it contains no misspelling.
- [x] The code is well commented, especially in the parts that contain more complexity.
- [ ] New and old tests passed locally.

## Reviewer's checklist

Copy and paste this template for your review's note:

```
## Reviewer's Checklist

- [ ] I managed to reproduce the problem locally from the `main` branch
- [ ] I managed to test the new changes locally
- [ ] I confirm that the issues mentioned were fixed/resolved .
```

### CODE REVIEW

**Comment by tintayadev** [comment]
Please, don't merge it yet. We are testing the CI.

**Comment by felipepaes** [comment]
Thank you for working on that! @tintayadev 
@xmnlab could you approve the workflow to run so we can check if @tintayadev changes fixed the issue, please?

File: conda/dev.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ channels:
   - nodefaults
   - conda-forge
 dependencies:
-  - python >=3.8.1
+  - python >=3.8.1, <=3.11.5
```
Comment (tintayadev) [inline conda/dev.yaml:6]: This was the line that was allowing conda to install python 3.12 in the environment

**Comment by xmnlab** [comment]
> Thank you for working on that! @tintayadev @xmnlab could you approve the workflow to run so we can check if @tintayadev changes fixed the issue, please?

done! btw, I just added you as maintainer in this repo and I send you an invitation to osl-incubator as well (I really thought that I sent that before .. but I was wrong XD)

**Pull request merged**